packages/compiler/src/types.ts

```diff
@@ -0,0 +1,45 @@
+export type StaticValue =
+  | string
+  | number
+  | boolean
+  | undefined
+  | Array<string | number | undefined>
+  | {
+      [_ in string]: StaticValue;
+    };
+
+export type Conditional = {
+  type: "Conditional";
+  expression: string;
+  whenTrue: StaticValue;
+  whenFalse: StaticValue;
+};
+
+export type Value =
```

Comment:
Note: This is the most important type in this PR, which tries to encode the potential patterns (e.g. we see that conditional can only contain `StaticValue`, which means that nested conditionals are not allowed)